addons/account_edi/models/account_move.py

```diff
@@ -48,8 +227,69 @@ def message_post(self, **kwargs):
-            break
+            invoice = edi_formats._update_invoice_from_attachment(attachment, self)
+            if invoice:
+                break
+
+        return res
+
+    ####################################################
+    # Business operations
+    ####################################################
+
+    def action_process_edi_web_services(self):
+        self.edi_document_ids.filtered(lambda d: d.state in ('to_send', 'to_cancel'))._process_documents_web_services()
+
+
+class AccountMoveLine(models.Model):
+    _inherit = 'account.move.line'
+
+    ####################################################
+    # Export Electronic Document
+    ####################################################
+
+    def reconcile(self):
+        # OVERRIDE
+        # In some countries, the payments must be sent to the government under some condition. One of them could be
+        # there is at least one reconciled invoice to the payment. Then, we need to update the state of the edi
+        # documents during the reconciliation.
+        all_lines = self + self.matched_debit_ids.debit_move_id + self.matched_credit_ids.credit_move_id
+        payments = all_lines.move_id.filtered(lambda move: move.payment_id or move.statement_line_id)
+
+        invoices_per_payment_before = {pay: pay._get_reconciled_invoices() for pay in payments}
+        res = super().reconcile()
+        invoices_per_payment_after = {pay: pay._get_reconciled_invoices() for pay in payments}
+
+        changed_payments = self.env['account.move']
+        for payment, invoices_after in invoices_per_payment_after.items():
+            invoices_before = invoices_per_payment_before[payment]
+
+            if set(invoices_after.ids) != set(invoices_before.ids):
+                changed_payments |= payment
+        changed_payments._update_payments_edi_documents()
+
+        return res
+
+    def remove_move_reconcile(self):
+        # OVERRIDE
+        # When a payment has been sent to the government, it usually contains some information about reconciled
+        # invoices. If the user breaks a reconciliation, the related payments must be cancelled properly and then, a new
+        # electronic document must be generated.
+        all_lines = self + self.matched_debit_ids.debit_move_id + self.matched_credit_ids.credit_move_id
+        payments = all_lines.move_id.filtered(lambda move: move.payment_id or move.statement_line_id)
+
+        invoices_per_payment_before = {pay: pay._get_reconciled_invoices() for pay in payments}
+        res = super().remove_move_reconcile()
+        invoices_per_payment_after = {pay: pay._get_reconciled_invoices() for pay in payments}
+
+        changed_payments = self.env['account.move']
+        for payment, invoices_after in invoices_per_payment_after.items():
+            invoices_before = invoices_per_payment_before[payment]
+
+            if set(invoices_after.ids) != set(invoices_before.ids):
```

Comment:
recordsets does not need set()

Reply:
I prefer like that since `.ids` is returning a `list` and not a `set`. Then, `env[model].browse([1, 2]).ids != env[model].browse([2, 1]).ids`